src/smexperiments/trial_component.py

```diff
@@ -92,8 +92,23 @@ def save(self):
         """Save the state of this TrialComponent to SageMaker."""
         return self._invoke_api(self._boto_update_method, self._boto_update_members)
 
-    def delete(self):
-        """Delete this TrialComponent from SageMaker."""
+    def delete(self, disassociate=None):
+        """Delete this TrialComponent from SageMaker.
+
+        Args:
+            disassociate (boolean): Indicates whether to disassociate the trial component with the trials before
+            deletion. If set to true, disassociate the trial component with associated trials first, then delete
+             the trial component. If it's not set or set to false, it will delete the trial component directory
+             without disassociation.
+        """
+        if disassociate:
+            trials = self.sagemaker_boto_client.list_trials(TrialComponentName=self.trial_component_name)[
```

Comment:
Pagination?

Reply:
Thoughts here? Does the boto client paginate for you?

Reply:
based on https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html?highlight=list_trials#SageMaker.Client.list_trials, boto client support pagination with NextToken. I will fix that in the next pr